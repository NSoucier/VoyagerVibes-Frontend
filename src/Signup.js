import React, { useState, useEffect } from "react";
import { Form, FormGroup, Label, Input, Button } from "reactstrap";
import { useNavigate } from "react-router-dom";
import './UserForms.css'

function Signup({ register, setCurrentUser }) {
    const [ form, setForm ] = useState({username: '', email: '', firstName:'', lastName: '', password: ''});
    const navigate = useNavigate();

    useEffect(() => {
        // if user is already logged in, redirect to profile page
        if (localStorage.user) navigate('/profile')
    }, []);

    // update signup-form data with user inputs
    function handleInput(evt) {
        evt.preventDefault();
        form[evt.target.name] = evt.target.value;
        setForm(form);
    }

    // submit form data to API and register user
    async function handleClick (evt) {
        evt.preventDefault();
        // show loading spinner
        const spinner = document.getElementById('buffering');
        spinner.innerHTML = '<img src="https://www.jsfirm.com/resources/ajax-loader.gif" id="spinner"/>';
        
        let status = await register(form);
        // if present, show errors
        if (!status.username) {
            const error = document.getElementById('error');
            error.innerText = status;
            spinner.innerHTML = '';
        } else {
            // set currentuser state
            setCurrentUser(status.username)
            // redirect to explore page upon successful registration
            navigate('/explore');
        }
    }

    return (
        <>
            <h2>Create an account</h2>
            <Form id="loginForm">
                <FormGroup floating>
                    <Input name="username" type="text" placeholder="" onChange={handleInput}/>
                    <Label for="username">Username</Label>
                </FormGroup>{" "}
                <FormGroup floating>
                    <Input name="email" type="email" placeholder="" onChange={handleInput}/>
                    <Label for="email">Email</Label>
                </FormGroup>{" "}
                <FormGroup floating>
                    <Input name="firstName" type="text" placeholder="" onChange={handleInput}/>
                    <Label for="firstName">First name</Label>
                </FormGroup>{" "}
                <FormGroup floating>
                    <Input name="lastName" type="text" placeholder="" onChange={handleInput}/>
                    <Label for="lastName">Last name</Label>
                </FormGroup>{" "}
                <FormGroup floating>
                    <Input name="password" type="password" placeholder="" onChange={handleInput}/>
                    <Label for="password">Password</Label>
                </FormGroup>{" "}            
                <p id="error" style={{color: 'red'}}></p>
                <Button color="success" onClick={handleClick}>Submit</Button>
                <div id="buffering"></div>
            </Form>
      </>
    )
}

export default Signup;