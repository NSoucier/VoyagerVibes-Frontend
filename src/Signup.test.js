import React, { act } from "react";
import { render, fireEvent } from "@testing-library/react";
import Signup from "./Signup";
import { Route, Routes, MemoryRouter, BrowserRouter } from "react-router-dom";

// smoke test
it("renders without crashing", async function () {
    await act(() => {
        render(
            <BrowserRouter>
                <Signup />
            </BrowserRouter>
          );
    });
});

// snapshot test
it("matches snapshot", function() {
    const {asFragment} = render(
        <BrowserRouter>
            <Signup />
        </BrowserRouter>
      );
    expect(asFragment()).toMatchSnapshot();
});
