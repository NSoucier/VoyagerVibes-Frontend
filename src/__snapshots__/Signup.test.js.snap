// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <h2>
    Create an account
  </h2>
  <form
    class=""
    id="loginForm"
  >
    <div
      class="mb-3 form-floating"
    >
      <input
        class="form-control"
        name="username"
        placeholder=""
        type="text"
      />
      <label
        class="form-label"
        for="username"
      >
        Username
      </label>
    </div>
     
    <div
      class="mb-3 form-floating"
    >
      <input
        class="form-control"
        name="email"
        placeholder=""
        type="email"
      />
      <label
        class="form-label"
        for="email"
      >
        Email
      </label>
    </div>
     
    <div
      class="mb-3 form-floating"
    >
      <input
        class="form-control"
        name="firstName"
        placeholder=""
        type="text"
      />
      <label
        class="form-label"
        for="firstName"
      >
        First name
      </label>
    </div>
     
    <div
      class="mb-3 form-floating"
    >
      <input
        class="form-control"
        name="lastName"
        placeholder=""
        type="text"
      />
      <label
        class="form-label"
        for="lastName"
      >
        Last name
      </label>
    </div>
     
    <div
      class="mb-3 form-floating"
    >
      <input
        class="form-control"
        name="password"
        placeholder=""
        type="password"
      />
      <label
        class="form-label"
        for="password"
      >
        Password
      </label>
    </div>
     
    <p
      id="error"
      style="color: red;"
    />
    <button
      class="btn btn-success"
      type="button"
    >
      Submit
    </button>
  </form>
</DocumentFragment>
`;
